pycloudlib/ec2/vpc.py

```diff
@@ -11,25 +11,30 @@
 class VPC:
     """Virtual Private Cloud Class."""
 
-    def __init__(self, resource, name, ipv4_cidr='192.168.1.0/20'):
+    def __init__(self, resource, name, ipv4_cidr='192.168.1.0/20',
+                 vpc_id=None):
         """Define and setup a VPC.
 
         Args:
             resource: EC2 resource object
             name: name of VPC
             ipv4_cidr: CIDR for IPV4 network
+            vpc_id: Optional ID of existing VPC object to return
```

Comment:
great suggestion @OddBloke. Thanks for the review. I refactored the class to make it a cleaner proxy which also lead me to drop carrying around a instance _resource attribute as well. Basically these proxy classes can be very simple and provide a couple helper methods to setup,  teardown, tagging and attribute lookup.

Reply:
@OddBloke +1. Yes this involved a bit of a refactor on the class as a whole as each private method was making using use of both the instance attribute self._resource and self.name in many places. But, I think as a whole it simplifies the proxy class   a bit since resource, logger and name are mostly used during creation and not runtime the majority of CI runtime (where the vpc would have already been created).